kroxylicious-filters/kroxylicious-encryption/src/main/java/io/kroxylicious/filter/encryption/inband/KeyContext.java

```diff
@@ -67,14 +74,19 @@ public int encodedSize(int plaintextSize) {
      * @param output The output buffer
      */
     public void encode(@NonNull ByteBuffer plaintext, @NonNull ByteBuffer output) {
-        if (remainingEncryptions.get() < 0) {
+        if (remainingEncryptions <= 0) {
             throw new ExhaustedDekException("No more encryptions");
         }
         encryptor.encrypt(plaintext, output);
     }
 
     @Override
     public void destroy() throws DestroyFailedException {
+        alive = false;
         encryptor.destroy();
     }
+
+    public boolean isAlive() {
+        return alive;
```

Comment:
Would isExpired be a better name?

Reply:
went for implementing `Destroyable` completely and using `isDestroyed`